app_dart/lib/src/service/scheduler.dart

```diff
@@ -369,7 +392,15 @@ class Scheduler {
       repository: pullRequest.base!.repo!.fullName,
       sha: pullRequest.head!.sha,
     );
-    final CiYaml ciYaml = await getCiYaml(commit);
+    late CiYaml ciYaml;
```

Comment:
Generally, you want to avoid late as loosens the restrictions on the compiler and can lead to bugs. We can rework this to instead have a nullable toCiYaml

```dart
CiYaml? totCiYaml;
if (commit.branch == Config.defaultBranch(commit.slug)) {
  final Commit totCommit = await generateTotCommit(slug: commit.slug, branch: Config.defaultBranch(commit.slug));
  totCiYaml = await getCiYaml(totCommit);
}
final CiYaml ciYaml = getCiYaml(commit, totCiYaml: totCiYaml);
```

Reply:
this is very artful! impressive!